projects/sol/src/lib/sol.component.spec.ts

```diff
@@ -22,13 +22,29 @@ describe('SolComponent', () => {
     .compileComponents()
   })
 
-  beforeEach(() => {
+  beforeEach(fakeAsync(() => {
     fixture = TestBed.createComponent(SolComponent)
     component = fixture.componentInstance
+    component.mpsServer = 'wss://localhost'
+    component.authToken = 'authToken'
+    tick(4500)
     fixture.detectChanges()
-  })
+    flush()
+  }))
 
   it('should create', () => {
     expect(component).toBeTruthy()
   })
+
+  it('should clear the core objects on sol stop', () => {
+    component.stopSol()
+
+    expect(component.term).toBe(null)
```

Comment:
This is missing a lot of expectations:
what if redirector is null
how do you know the redirector.stop method was called?
how do know that the terminal was reset method was called and that it was disposed?
what about redirector, dataProcessor and term being set to null?

Reply:
Updated the expectations.